inspirehep/modules/workflows/tasks/merging.py

```diff
@@ -48,3 +59,60 @@ def get_head_source(head_uuid):
         return None
 
     return 'arxiv' if 'arxiv' in roots_sources else 'publisher'
+
+
+@with_debug_logging
+def has_conflicts(obj, eng):
+    """Return if the workflow has any confilicts."""
+    return obj.extra_data.get('conflicts')
+
+
+@with_debug_logging
+def merge_articles(obj, eng):
+    """Merge two articles.
+
+    The workflow payload is overwritten by the merged record, the conflicts are
+    stored in ``extra_data.conflicts``. Also, it adds a ``callback_url`` which
+    contains the endpoint which resolves the merge conflicts.
+
+    Note:
+        For the time being the ``root`` will be ignored, and we'll rely only
+        on the ``head``, hence it is a rootless implementation.
+
+    Returns:
+        bool: ``True`` if the workflow merged with another record in the system.
+        ``False`` if the merger feature is not enabled.
+
+    """
+    if not current_app.config.get('FEATURE_FLAG_ENABLE_MERGER'):
+        return None
+
+    if obj.extra_data.get('fuzzy-matched') is True:
+        matched_control_number = get_value(
+            obj.extra_data, 'fuzzy_match_approved_id')
+    else:
+        matched_control_number = get_value(obj.extra_data, 'matches.exact[0]')
```

Comment:
`matches.approved` is the place where you will find the matched record, whether it was a fuzzy or an exact match, thus you don't need the `if-else`statement, just assume it is there.
I would even access it using `obj.extra_data.['matches']['approved']`, in case it blows up if no match is set